packages/lib/BaseApplication.ts

```diff
@@ -636,6 +637,11 @@ export default class BaseApplication {
 		// https://immerjs.github.io/immer/docs/freezing
 		setAutoFreeze(initArgs.env === 'dev');
 
+		// Work around issues with ipv6 resolution
+		// (possibly incorrect URL serialization see https://github.com/mswjs/msw/issues/1388#issuecomment-1241180921).
+		// See also https://github.com/node-fetch/node-fetch/issues/1624#issuecomment-1407717012
+		dns.setDefaultResultOrder('ipv4first');
+
```

Comment:
We might want to add an option to disable this &mdash; `setDefaultResultOrder` has a [higher precedence than `NODE_OPTIONS`](https://nodejs.org/api/dns.html#dnssetdefaultresultorderorder) and users could previously enable/disable this with `NODE_OPTIONS=--dns-result-order=verbatim` or `NODE_OPTIONS=--dns-result-order=ipv4first`.[^1]

[^1]: Suggested by https://github.com/nodejs/node/issues/41145#issuecomment-991929539

Reply:
The pull request has been updated to not override the default result order if `--dns-result-order=something` is specified on the command line.